drawbicyc/include/drawbicyc/plot_series.hpp

```diff
@@ -594,12 +643,68 @@ void plot_series(Plotter_t plotter, Plots plots, std::string type)
         {
           {
             typename Plotter_t::arr_t snap(plotter.h5load_rr_timestep(at * n["outfreq"]));
-            snap *= rhod * 1e3; // water per cubic metre (should be wet density...) & g/kg
+            snap *= rhod; //kg/kg * 1e3; // water per cubic metre (should be wet density...) & g/kg
             res_prof(at) = blitz::mean(snap); 
           }
         }
         catch(...) {;}
-      }   
+      }
+      else if (plt == "cwp")
+      {
+        // cloud water path
+        try
+        {
+          {
+            typename Plotter_t::arr_t snap(plotter.h5load_rc_timestep(at * n["outfreq"]));
+            snap *= rhod; //kg/kg *1e3; // water per cubic metre (should be wet density ...) & g/kg
+            res_prof(at) = blitz::mean(snap);
+          }
+        }
+        catch(...) {;}
+      }
+      else if (plt == "lwm")
+      {
+        //liquid water mass
+        try
+        {
+          { 
+            auto tmp = plotter.h5load_rc_timestep(at * n["outfreq"]);
+            typename Plotter_t::arr_t snap(tmp);
+            snap += plotter.h5load_rr_timestep(at * n["outfreq"]);
+            snap *= rhod;
+            res_prof(at) = blitz::sum(snap) * plotter.DomainSurf;
```

Comment:
Why multiply by DomainSurf? 
Together with multiplication by domain height, which is done later, this results in multiplying by domain volume.
This would work if you calculated mean of snap, not sum of it.
However, then it is more difficult to precisly account for smaller cell on top and bottom.

I think that it would be easier and more readable to multiply snap by cell volume, to get total mass of liquid water in a cell. Then, divide uppermost and lowermost horizontal slices of the snap array by 2 to account for smaller cells there and finally to calculate result as a sum of snap.

Same goes for cwm and rwm.